job/job.go

```diff
@@ -110,14 +113,45 @@ func (j *Job) UnmarshalJSON(b []byte) error {
 	j.AggrID = aggrID
 
 	cbURL, ok := tmp["callback_url"].(string)
-	if !ok {
-		return errors.New("callback_url must be a string")
+	if ok {
+		_, err = url.ParseRequestURI(cbURL)
+		if err != nil {
+			return errors.New("Could not parse callback URL: " + err.Error())
+		}
+		j.CallbackURL = cbURL
 	}
-	_, err = url.ParseRequestURI(cbURL)
-	if err != nil {
-		return errors.New("Could not parse callback URL: " + err.Error())
+
+	// Check if callback_type and callback_dst are present
+	// only if callback_url is empty
+	if j.CallbackURL == "" {
+		cbType, typeOK := tmp["callback_type"].(string)
+		cbDestination, dstOK := tmp["callback_dst"].(string)
+
+		if cbType == "" && cbDestination == "" {
+			return errors.New("You need to provide either callback_url OR callback_type and callback_dst. These parameters are empty!")
+		} else if cbType != "" && cbDestination == "" {
+			return errors.New("callback_dst cannot be empty")
+		} else if cbType == "" && cbDestination != "" {
+			return errors.New("callback_type cannot be empty")
+		}
+
+		if !typeOK {
+			return errors.New("callback_type must be a string!")
+		}
+
+		if !dstOK {
+			return errors.New("callback_dst must be a string!")
+		}
```

Comment:
These two checks should be moved right after `typeOK` and `dstOK` are assigned, respectively.